reyna/src/com/b2msolutions/reyna/services/StoreService.java

```diff
@@ -38,28 +40,44 @@ public static void setCellularDataBlackout(Context context, TimeRange range) {
         new Preferences(context).saveCellularDataBlackout(range);
     }
 
-	@Override
-	protected void onHandleIntent(Intent intent) {
+    public static void setStorageSizeLimit(long limit) {
+        StoreService.storageSizeLimit = limit;
```

Comment:
This need to be in a persist storage like a shared-Preference.
for many reasons, most important one is that if device rebooted or process crash then an event immediately occurred then the size will be unlimited.

@Youhana-Hana